### PR TITLE
Updates for retry changes in 0.6.

### PR DESCRIPTION
**Pull request by rofinn**
I keep including the same retry version check in my code, so I figured this made more sense.

### CODE REVIEW

**Comment by rofinn** [comment]
Looks like the kwargs support to retry won't work on 0.4. Would it make sense to drop support for 0.4 now that we're moving on to 0.7?

**Comment by tkelman** [comment]
now's about the right time for that, yes

**Comment by rofinn** [comment]
I'm guessing that should be a separate PR which removes any 0.4 specific code?

**Comment by tkelman** [comment]
yeah, and deprecations that I know @timholy was wanting to get rid of or adjust (hopefully the consumers of them have moved on to not using them from Compat any more...)

**Comment by rofinn** [comment]
Should be good to review now that 0.4 has been dropped.

**Comment by tkelman** [comment]
Needs a README mention

**Comment by ararslan** [comment]
Bump. Needs conflicts resolved then should be good to go.

**Comment by rofinn** [comment]
Should be good to go now.

File: README.md [hunk 1]
```diff
@@ -155,11 +155,12 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `logdet` for `Number`s ([#22629]).
 
-* `fieldcount` is equivalent to `nfields` for Julia versions 0.6 and below and is used to
-  determine the number of fields in a data type ([#22350]).
+* `fieldcount` is equivalent to `nfields` for Julia versions 0.6 and below and is used to determine the number of fields in a data type ([#22350]).
```
Comment (ararslan) [inline README.md]: Unrelated change?

Comment (rofinn) [inline README.md]: Yeah, it's just a new line change I accidentally introduced as part of resolving conflicts. Didn't think it was a big issue.

Comment (tkelman) [inline README.md]: would be best not to touch this as it would be misleading in blame of the readme

**Comment by rofinn** [comment]
Bump?
